src/protocols/permissions.ts

```diff
@@ -66,6 +66,10 @@ export type PermissionRevocationCreateOptions = {
    */
   signer?: Signer;
   grantId: string;
+  /**
+   * If the grant being revoked was scoped to a protocol, the protocol must be included in the revocation.
+   */
+  protocol?: string;
```

Comment:
Appreciate the additional documentation for the non-trivial requirement here! I wonder if it would be an improvement if we pass in the grant itself instead of the `grantId` so that this info can be inferred?

It is almost guaranteed that a dev is not gonna know they need to pass in protocol for protocol scoped grants.

And I think it is a valid assumption that surely whomever is revoking the grant has the grant they want to revoke, because how else would they know what the `protocol` value is to pass in LOL!

Totally fine if you want to address this in a separate PR even if you agree, I try to avoid bloating existing PRs myself!

Reply:
That's a good call, I like it! I'll do it in this PR just so that we actually get it done, i think it's better devex.

Reply:
Done